File: BAEKJOON/cpp/20112_샤토르_마방진.cpp
#include <iostream>
#include <vector>
using namespace std;

int n;
vector<vector<char>> v;

void input() {
	cin >> n;

	v.assign(n, vector<char>(n));

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			cin >> v[i][j];
		}
	}
}

void solution() {
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			if (v[i][j] != v[j][i]) {
				cout << "NO";
				return;
			}
		}
	}

	cout << "YES";
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}

File: BAEKJOON/java/20112_샤토르_마방진.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static List<List<Character>> map = new ArrayList<>();

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String s = br.readLine();
            List<Character> temp = new ArrayList<>();

            for (int j = 0; j < s.length(); j++) {
                temp.add(s.charAt(j));
            }

            map.add(temp);
        }
    }

    private static void solution() throws IOException {
        for (int i = 0; i < map.size(); i++) {
            for (int j = 0; j < map.get(i).size(); j++) {
                if (map.get(i).get(j) != map.get(j).get(i)) {
                    System.out.println("NO");
                    return;
                }
            }
        }

        System.out.println("YES");
    }
}